bzt/modules/aggregator.py

```diff
@@ -181,6 +185,39 @@ def recalculate(self):
 
         return self
 
+    def _compact_times(self):
+        times = self[KPISet.RESP_TIMES]
+        redundant_cnt = len(times) - self.rt_dist_maxlen
+        if redundant_cnt > 0:
+            logging.debug("Compacting %s response timing into %s", len(times), self.rt_dist_maxlen)
+
+        while redundant_cnt > 0:
+            keys = sorted(times.keys())
+            distances = [(lidx, keys[lidx + 1] - keys[lidx]) for lidx in range(len(keys) - 1)]
+            distances.sort(key=operator.itemgetter(1))  # sort by distance
+
+            # cast candidates for consolidation
+            lkeys_indexes = [lidx for lidx, _ in distances[:redundant_cnt]]
+
+            while lkeys_indexes:
+                lidx = lkeys_indexes.pop(0)
+                lkey = keys[lidx]
+                rkey = keys[lidx + 1]
+                if lkey in times and rkey in times:  # neighbours aren't changed
+                    lval = times.pop(lkey)
+                    rval = times.pop(rkey)
+
+                    # shift key proportionally to values
+                    idx_new = lkey + (rkey - lkey) * float(rval) / (lval + rval)
+
+                    # keep precision the same
+                    lprec = len(str(math.modf(lkey)[0])) - 2
```

Comment:
We're rounding up the end value to stay with same precision as original values. Like if we had 0.111 and 0.333 and resulting value is something like 0.2333333, we round it into 0.233. Otherwise there is huge chance that next KPiset merge will bring more _new_ indexes.